packages/bun-types/sqlite.d.ts

```diff
@@ -487,6 +487,46 @@ declare module "bun:sqlite" {
      * @link https://www.sqlite.org/c3ref/file_control.html
      */
     fileControl(zDbName: string, op: number, arg?: ArrayBufferView | number): number;
+
+    /**
```

Comment:
`Database` isn't currently an `EventEmitter` or an `EventTarget`. I think it will be confusing to have these methods which look like one of those, but aren't

Instead, maybe we could have some getters/setters like

```
database.onUpdate = (database, tableName, rowId) => {}
database.onInsert = (database, tableName, rowId) => {}
database.onDelete = (database, tableName, rowId) => {}
```

Then, in sqlite.ts we could create these functions via .bind, so that the first argument is always the `database` object and then the `tableName` and `rowId` are passed along

These would be implemented as getter/setters on the `Database` instance itself, and unregistered when the finalization callback occurs in the JSSQLDatabase probably.

Reply:
The other option is we make `Database` an `EventEmitter` and that would also be an answer to how it would handle exceptions and promise rejections in these callbacks.

Reply:
> Instead, maybe we could have some getters/setters like

This approach would prevent us from setting multiple listeners, wouldn't it?
Maybe this whole logic should not be part of sqlite.ts and we can just expose the `updateHook` function with documentation and let people build their event emitter on top of it?